open_instruct/grpo_fast.py

```diff
@@ -1574,11 +1591,50 @@ def accumulate_inference_batches(
 
         query, ground_truth, dataset, raw_query = pending_queries_map.pop(result.dataset_index)
 
+        for i in range(len(result.finish_reasons)):
+            if result.finish_reasons[i] == "stop" and len(result.responses[i]) == 0:
+                result.responses[i].append(tokenizer.eos_token_id)
+                result.masks[i].append(1)
+                result.logprobs[i].append(float("nan"))
+
+        decoded_responses = tokenizer.batch_decode(result.responses, skip_special_tokens=True)
+
+        k_queries = [query for _ in range(generation_config.n)]
+        k_ground_truths = [ground_truth for _ in range(generation_config.n)]
+        k_datasets = [dataset for _ in range(generation_config.n)]
+        k_decoded_queries = [raw_query for _ in range(generation_config.n)]
+
+        with Timer("💰 [Data Preparation Thread] Calculating rewards and advantages"):
+            scores, reward_metrics = asyncio.run(
+                reward_fn(
+                    result.responses,
+                    decoded_responses,
+                    Batch(ground_truths=k_ground_truths, datasets=k_datasets),
+                    result.finish_reasons,
+                    result.request_info,
+                    k_decoded_queries,
+                )
+            )
+
+        # Filter out zero std prompts
+        if filter_zero_std_samples and np.array(scores).std() == 0:
```

Comment:
We can just do `np.std(scores)` without the array!

Reply:
done!